cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -168,9 +188,59 @@ private Process launchCassandraNodeTask(@NotNull final Marker taskIdMarker, @Not
 
     @NotNull
     private CassandraNodeHealthCheckDetails performHealthCheck(@NotNull final CassandraNodeHealthCheckTask healthCheckTask) {
-        return CassandraNodeHealthCheckDetails.newBuilder()
-            .setHealthy(true)
-            .build();
+        CassandraNodeHealthCheckDetails.Builder builder = CassandraNodeHealthCheckDetails.newBuilder();
+        try (JmxConnect jmx = new JmxConnect(getHostAddress(), (int) healthCheckTask.getJmxPort() & 0xffff)) {
```

Comment:
I assume the `0xffff` is to deal with the down casting from `long` to `int` and making sure that it's a valid positive value?